src/realm/query_conditions.hpp

```diff
@@ -240,6 +274,17 @@ struct EndsWith : public HackClass {
         return v2.ends_with(v1);
     }
 
+    bool operator()(const Mixed& m1, const Mixed& m2, bool = false, bool = false) const
```

Comment:
Maybe add some tests for EndsWith and Equal to make sure the behaviour is expected in regards to the end character being null or not. My intuition is that if people are trying to compare String vs Binary, they would only compare equal if both contain the trailing null, but I think we are free to define this if you feel otherwise.

Reply:
Damn this is tricky.